### PR TITLE
Add profile ui and user entity

### PR DESCRIPTION
**Pull request by scogra17**
Tested both locally and with `heroku local`.

There are several changes here, which I'd normally break up into several smaller PRs, but I got a little too eager :) 

Changes include: 
1) Added temporary button to index page for testing (makes a GET request for a particular users profile)
1) Added profile UI (will eventually only be accessible to logged in users)
2) Added layout to make footer re-usable
3) Added user entity
4) Added database method to query for a user by ID

Screenshots: 
Screen Shot 2021-12-29 at 4 07 38 PM
Screen Shot 2021-12-29 at 4 07 32 PM

### CODE REVIEW

File: user.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class User
+  attr_reader :slackname, :firstname, :lastname, :about_me, :track, :course, :timezone
+
+  def initialize(slackname, firstname, lastname, about_me, track, course, timezone)
+    @slackname = slackname
+    @firstname = firstname
+    @lastname = lastname
+    @about_me = about_me
+    @track = track
```
Comment (scogra17) [inline user.rb:9]: I've defined the track, course and timezone instance variables as text. They could just as easily be their own classes ("entities" ), but I thought that might be overkill for now.

File: database_persistence.rb [hunk 2]
```diff
@@ -17,6 +19,28 @@ def query(statement, *params)
...
+        ti.name AS timezone
+      FROM users u
+        JOIN tracks t ON u.track_id = t.id
+        JOIN courses c ON u.course_id = c.id
+        JOIN timezones ti ON u.timezone_id = ti.id
+      WHERE u.id = $1
+    SQL
+
+    result = query(sql, id)
+    tuple = result.first
```
Comment (AlonsoLobato) [inline database_persistence.rb:40]: I don't understand why we need to call `first` here; isn't the `query` method already returning just one element, the row in the db which `id` is passed to `query`, right?

File: peers.rb [hunk 3]
```diff
@@ -24,3 +25,10 @@
   @user = @storage.first_user()
   return "#{@user["firstname"]} (id: #{@user["pkey"]}) is a user"
 end
```
Comment (AlonsoLobato) [inline peers.rb:27]: This route as defined here is still just used for testing purposes, right? Is just retrieves the first used in the db, correct?

File: test_data.sql [hunk 4]
```diff
@@ -18,7 +18,7 @@ INSERT INTO users (id, firstname, lastname, slackname, email, password,
   VALUES ('fc10b881-d9a0-4ab1-a6fd-a102db188f49', 'Scott', 'Graham', 'Scott', 'scttgrhm7@gmail.com', 'ldfgkj78%^&appdKO039*',
           2, 1, 3, 'Finishing LS and planning to do capstone in May. I like surfing and thick grey socks'),
          ('fc10b881-d9a0-4ab1-a6fd-a102db188f50', 'Alonso', 'Lobato', 'Alonso Lobato', 'alonzilj@gmail.com', 'ppdK78%^&aO039*ldfgkj',
-          1, 1, 1, 'About to finish back-end portion of Ruby track. I like canadian Christmas trees');
+          1, 1, 1, 'Finished back-end portion of Ruby track! I like canadian Christmas trees');
```
Comment (AlonsoLobato) [inline test_data.sql:21]: Nice catch! :)

File: user.rb (hunk 1 above)
Comment (AlonsoLobato) [reply user.rb:9]: I need further explain on this comment...

File: database_persistence.rb [hunk 5]
```diff
@@ -17,6 +19,28 @@ def query(statement, *params)
...
+    tuple_to_user(tuple)
+  end
+
   def first_user()
```
Comment (AlonsoLobato) [inline database_persistence.rb:44]: Same comment as in peers.rb file. This method is for testing purposes only, right?

File: views/footer_layout.erb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
```
Comment (AlonsoLobato) [inline views/footer_layout.erb:1]: I'd like to discuss the purpose of this layout

File: peers.rb [hunk 7]
```diff
@@ -24,3 +25,10 @@
...
+
+get "/test/profile/:id" do
+  user_id = params[:id]
+  @user = @storage.user_by_id(user_id)
+
+  erb :profile, layout: :footer_layout
```
Comment (AlonsoLobato) [inline peers.rb:33]: I'd like to discuss the purpose of `:footer_layout` layout

**Review by AlonsoLobato — APPROVED** [review]
Fantastic job man! Thanks again. I left some comments in several files. Nothing to review or change, just questions I've got about "this and that" that I hope we can discuss later today. By the way, I hope the comments remain after I approve this PR... Talk to you soon!

File: user.rb (hunk 1 above)
Comment (scogra17) [reply user.rb:9]:
```
  def tuple_to_user(tuple)
    slackname = tuple["slackname"]
    firstname = tuple["firstname"]
    lastname = tuple["lastname"]
    about_me = tuple["about_me"]
    track_code = tuple["track"]
    track_name = ...
    course = tuple["course"]
    timezone = tuple["timezone"]

    User.new(
      slackname,
      firstname,
      lastname,
      about_me,
      Track.new(track_code, track_name),
      Course.new(...),
      Timezone.new(...)
    )
  end
```

File: database_persistence.rb (hunk 2 above)
Comment (scogra17) [reply database_persistence.rb:40]: Yep, I'll delete first, but I'll have to test that the returned data type is still a tuple and an array of tuples of length one

Comment (scogra17) [reply database_persistence.rb:40]: Just kidding! exec_params always returns a collection and therefore requires selecting an index, even if the collection has a length of one

File: peers.rb (hunk 3 above)
Comment (scogra17) [reply peers.rb:27]: Yep!

File: database_persistence.rb (hunk 5 above)
Comment (scogra17) [reply database_persistence.rb:44]: Yep!

**Comment by scogra17** [comment]
Awesome. Thanks, Alonso!

**Pull request merged**